src/averageAges.js

```diff
@@ -15,11 +15,23 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  function menFilter(el) {
+    return el.sex === 'm';
+  }
+
+  function deathFilter(el) {
+    const deathCentury = Math.ceil(el.died / 100);
+
+    return century ? deathCentury === century : el.died !== century;
+  }
+
+  const men = people.filter(menFilter).filter(deathFilter);
```

Comment:
no need to filter two times just use different callbacks when century specified and if it's doesn't